### PR TITLE
[AIRFLOW-2213] Add Quoble check operator

### PR DESCRIPTION
**Pull request by sakshi2894**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2213

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

This extends CheckOperator and ValueCheckOperator for Qubole Commands.
It also adds another optional parameter called `results_parser_callable` which 
is a python_callable that should hold the logic to parse a list of rows returned by 
the Qubole command and return a list of records on which the checks have to be 
performed. By default, only the first row is parsed. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
- Adds qubole_check_operator tests to check 
       - if the operator passes if the query result is within a permissible range of values. 
       - if the operator fails if the query results is not with a permissible limit. 
       - if pass_value gets templatized properly.
       - if the operator fails, it contains Qubole command related information in the exception 
          message.
- Adds `qubole_check_hook` tests to test the default parse_first_row method in case of different query results.
 

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/operators/qubole_check_operator.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        :tags: array of tags to be assigned with the command
+        :cluster_label: cluster label on which the command will be executed
+        :name: name to be given to command
+        :notify: whether to send email on command completion or not (default is False)
+        :results_parser_callable: a python_callable which parses a list of rows
+        returned by the qubole command. By default, only the first
+        row is parsed. The python_callable should return a list of records
+        on which the checks have to be performed.
+
+        **Arguments specific to command types**
```
Comment (msumit) [inline airflow/contrib/operators/qubole_check_operator.py]: can't we just fallback to QuboleOperator's docstring for this?

File: airflow/contrib/operators/qubole_check_operator.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        sql = params['query']
+    elif 'sql' in params:
+        sql = params['sql']
+    return sql
+
+
+def handle_airflow_exception(airflow_exception, hook):
+    cmd = hook.cmd
+    if cmd is not None:
+        if cmd.status == 'done':
```
Comment (msumit) [inline airflow/contrib/operators/qubole_check_operator.py]: better to use sdk given method, i.e. https://github.com/qubole/qds-sdk-py/blob/master/qds_sdk/commands.py#L55

File: airflow/contrib/operators/qubole_check_operator.py [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    kwargs:
+        :command_type: type of command to be executed, e.g. hivecmd, shellcmd, hadoopcmd
+        :tags: array of tags to be assigned with the command
+        :cluster_label: cluster label on which the command will be executed
+        :name: name to be given to command
+        :notify: whether to send email on command completion or not (default is False)
+        :results_parser_callable: a python_callable which parses a list of rows
+        returned by the qubole command. By default, only the first
+        row is parsed. The python_callable should return a list of records
+        on which the checks have to be performed.
```
Comment (msumit) [inline airflow/contrib/operators/qubole_check_operator.py]: also, add an comment about `tolerance`

File: airflow/contrib/operators/qubole_check_operator.py [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    ui_fgcolor = '#000'
+
+    @apply_defaults
+    def __init__(self, qubole_conn_id="qubole_default", *args, **kwargs):
+        sql = get_sql_from_qbol_cmd(kwargs)
+        super(QuboleCheckOperator, self)\
+            .__init__(qubole_conn_id=qubole_conn_id, sql=sql, *args, **kwargs)
+        self.on_failure_callback = QuboleCheckHook.handle_failure_retry
+        self.on_retry_callback = QuboleCheckHook.handle_failure_retry
+        self.hook = QuboleCheckHook(context=None, *self.args, **self.kwargs)
```
Comment (msumit) [inline airflow/contrib/operators/qubole_check_operator.py]: get the hook out of init

**Comment by msumit** [comment]
@sakshi2894 plz add an example dag as well

**Comment by sakshi2894** [comment]
Here are the documentation screenshots: 

**QuboleCheckOperator:**
screen shot 2018-05-11 at 12 42 40 am

**QuboleValueCheckOperator:**
screen shot 2018-05-11 at 12 43 08 am

**Comment by devjyotipatra** [comment]
+1
Useful enhancement

**Comment by ghost** [comment]
LGTM

**Comment by kaxil** [comment]
@msumit Any comments?

**Pull request closed without merging**